pygmt/clib/session.py

```diff
@@ -154,7 +154,8 @@ def info(self):
             self._info = {  # pylint: disable=attribute-defined-outside-init
                 "version": self.get_default("API_VERSION"),
```

Comment:
From https://github.com/GenericMappingTools/pygmt/issues/1830#issue-1173180803, you said:

> 2\. The GMT version information `gmt: 6.4.0_434ed18_2022.03.11` is obtained by calling `gmt --version`. However, when multiple GMT versions are installed, the command `gmt` and the library `libgmt` may come from different GMT versions. For example, if I have GMT 6.3 in my conda environment and GMT dev in another installation path, PyGMT may load the GMT 6.3 library but find GMT dev comman, or vice versa, depending on the settings of PATH and GMT_LIBRARY_PATH. It causes some confusion because PyGMT reports inconsistent GMT versions. I think we should remove the `gmt: 6.4.0_434ed18_2022.03.11` information.

So we will only have `6.4.0` printed out. This is ok for most users who install the offficial release versions, but for those who install the `dev` versions, I think it's useful to have the git short-hash and date part too, so that we can know the exact GMT dev version we've installed.

Just as an idea, how about we check the two strings? If there are equal (e.g. `6.4.0` and `6.4.0`), we just print one. But if they are different, then print both? Alternatively, is there a way to get the exact version from the GMT API somehow?

Reply:
> From [#1830 (comment)](https://github.com/GenericMappingTools/pygmt/issues/1830#issue-1173180803), you said:
> 
> > 2. The GMT version information `gmt: 6.4.0_434ed18_2022.03.11` is obtained by calling `gmt --version`. However, when multiple GMT versions are installed, the command `gmt` and the library `libgmt` may come from different GMT versions. For example, if I have GMT 6.3 in my conda environment and GMT dev in another installation path, PyGMT may load the GMT 6.3 library but find GMT dev comman, or vice versa, depending on the settings of PATH and GMT_LIBRARY_PATH. It causes some confusion because PyGMT reports inconsistent GMT versions. I think we should remove the `gmt: 6.4.0_434ed18_2022.03.11` information.
> 
> So we will only have `6.4.0` printed out. This is ok for most users who install the offficial release versions, but for those who install the `dev` versions, I think it's useful to have the git short-hash and date part too, so that we can know the exact GMT dev version we've installed.

Yes, I agree.

> Just as an idea, how about we check the two strings? If there are equal (e.g. `6.4.0` and `6.4.0`), we just print one. But if they are different, then print both? 

We still have the same problem. When multiple GMT versions are installed, we may load the libgmt from GMT version A and call GMT CLI from GMT version B. 

> Alternatively, is there a way to get the exact version from the GMT API somehow?

I just checked the GMT source code. It's impossible to get the version string `6.4.0_434ed18_2022.03.11` from GMT API. 

Since we can know the GMT library path `library path: /Users/user/opt/GMT-master/lib/libgmt.dylib`, perhaps we can assume that the relative path to the GMT command is `{library path}/../bin/gmt`? Or we can ask GMT to report the version string `6.4.0_434ed18_2022.03.11` through GMT API?

Reply:
> > Alternatively, is there a way to get the exact version from the GMT API somehow?
> 
> I just checked the GMT source code. It's impossible to get the version string `6.4.0_434ed18_2022.03.11` from GMT API.
> 
> Since we can know the GMT library path `library path: /Users/user/opt/GMT-master/lib/libgmt.dylib`, perhaps we can assume that the relative path to the GMT command is `{library path}/../bin/gmt`? Or we can ask GMT to report the version string `6.4.0_434ed18_2022.03.11` through GMT API?

Getting it through the GMT API seems like the most robust solution, rather than having to guess where the actual GMT binary PyGMT is using is located on the filesystem. Not sure if this is possible though to get the git hash information in C?

Reply:
I refer to the git commit information often when debugging (while careful to not have GMT installed via conda), so I added this option to the GMT_Get_Default API function in https://github.com/GenericMappingTools/gmt/pull/6493 so that we can keep this information in show_versions.